File: https:/github.com/abhinavtripathi17/DSA-Leetcode.git/1-TwoSum/1-TwoSum.cpp
// Last updated: 8/4/2026, 7:26:56 PM
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int n = nums.size();
        map<int,int>mpp;

        for(int i = 0 ; i < n ; i++){
            int curr = nums[i];
            int rem = target - nums[i];

            if(mpp.find(rem) != mpp.end()){
                return {mpp[rem] , i};
            }

            mpp[nums[i]] = i;
        }
        return {};
    }
};

File: https:/github.com/abhinavtripathi17/DSA-Leetcode.git/1-TwoSum/1-TwoSum.java
// Last updated: 8/4/2026, 7:27:00 PM
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer , Integer>mpp = new HashMap<>();
        for(int i = 0 ; i < nums.length ; i++){
            int req = target - nums[i];
            if(mpp.containsKey(req)){
                return new int [] {i,mpp.get(req)};
            }
            mpp.put(nums[i] , i);
        }
        return new int[]{};
    }  
    
}